tests/core/server/test_rate_limits.py

```diff
@@ -1,28 +1,46 @@
 import asyncio
+from typing import List
 
 import pytest
 
 from chia.protocols.protocol_message_types import ProtocolMessageTypes
+from chia.protocols.shared_protocol import Capability
 from chia.server.outbound_message import make_msg
-from chia.server.rate_limits import RateLimiter, NON_TX_FREQ
+from chia.server.rate_limit_numbers import get_rate_limits_to_use
+from chia.server.rate_limits import RateLimiter
+from chia.server.server import ChiaServer
+from chia.server.ws_connection import WSChiaConnection
+from chia.types.peer_info import PeerInfo
+from chia.util.ints import uint16
+from tests.conftest import node_with_params
 from tests.setup_nodes import test_constants
 
 
 constants = test_constants
+rl_v2 = [Capability.BASE, Capability.BLOCK_HEADERS, Capability.RATE_LIMITS_V2]
+rl_v1 = [Capability.BASE]
+node_with_params_b = node_with_params
+test_different_versions_results: List[int] = []
 
 
 class TestRateLimits:
+    @pytest.mark.asyncio
+    async def test_get_rate_limits_to_use(self):
+        assert get_rate_limits_to_use(rl_v2, rl_v2) != get_rate_limits_to_use(rl_v2, rl_v1)
+        assert get_rate_limits_to_use(rl_v1, rl_v1) == get_rate_limits_to_use(rl_v2, rl_v1)
+        assert get_rate_limits_to_use(rl_v1, rl_v1) == get_rate_limits_to_use(rl_v1, rl_v2)
+
     @pytest.mark.asyncio
     async def test_too_many_messages(self):
```

Comment:
it might be appropriate to have a fixture for rate-limit-version, that's parameterized on 1 and 2. similar to `db_version`

Reply:
The rate limits are different though so many tests would fail and require more significant changes. We do have an integration test at the bottom of the file though.